src/core/server/http/router/request.ts

```diff
@@ -161,6 +178,12 @@ export class KibanaRequest<
     // until that time we have to expose all the headers
     private readonly withoutSecretHeaders: boolean
   ) {
+    // The `requestId` property will not be populated for requests that are 'faked' by internal systems that leverage
+    // KibanaRequest in conjunction with scoped Elaticcsearch and SavedObjectsClient in order to pass credentials.
+    // In these cases, the id defaults to `internal`.
+    // This should be solved as part of https://github.com/elastic/kibana/issues/39430.
+    this.id = (request.app as KibanaRequestState | undefined)?.requestId ?? 'internal';
```

Comment:
How would we prefer to track 'fake requests' initiated by the system in this initial iteration? Right now I'm falling back to a `internal` id, but we could generate a new UUID here or leave it `undefined`.

cc @thomheymann

Reply:
Hi, thanks for pinging me! I think it would be good to have an id generated for fake requests too so that any subsequent operations and calls to elastic search can be correlated.

Reply:
One thing to note is that if a backend reuses a KibanaRequest object for multiple, separate invocations of a job, then this UUID will be re-used multiple times.

I haven't found any cases where we reuse the same KibanaRequest object. Could these teams confirm in their usages that we do not reuse the same KibanaRequest object?
@elastic/kibana-alerting-services @elastic/kibana-reporting-services

Reply:
From alerting's perspective, we build a single fake request for each task execution (job).